MyLinkedList.java

```diff
@@ -1 +1,147 @@
+package com.company;
 
+public class MyLinkedList<T> implements List<T> {
+
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+    private StringBuilder stringBuilder = new StringBuilder();
+
+    @Override
+    public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            first = newNode;
+            last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        }
+        size++;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0) {
+            add(value);
+        }
+        if (index == size) {
+            last.next = newNode;
+            last = newNode;
+        }
+        Node<T> leftNode = currentNode(index);
+        Node<T> rightNode = leftNode.prev;
+        leftNode.next = newNode;
+        rightNode.prev = newNode;
+        newNode.prev = leftNode;
+        newNode.next = rightNode;
+        size++;
+    }
+
+    @Override
+    public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+    }
+
+    @Override
+    public T get(int index) {
+        if (checkIndex(index)) {
+            return currentNode(index).item;
+        }
+        return null;
+    }
+
+    @Override
+    public void set(T value, int index) {
+        if (checkIndex(index)) {
+            currentNode(index).item = value;
+        }
+    }
+
+    @Override
+    public T remove(int index) {
+        if (checkIndex(index) && !isEmpty()) {
+            Node<T> removeNode = currentNode(index);
+            T value = removeNode.item;
+            if (removeNode == last) {
+                removeNode.prev = last;
+            } else if (removeNode == first) {
+                removeNode.next = first;
```

Comment:
the same

Reply:
I did it after 'if ... else if... else'
removeNode = new Node<>(null, null, null);
And maybe do you mean something else?